src/TokenManagers/XboxTokenManager.js

```diff
@@ -24,34 +30,28 @@ class XboxTokenManager {
     this.headers = { 'Cache-Control': 'no-store, must-revalidate, no-cache', 'x-xbl-contract-version': 1 }
   }
 
-  async getCachedUserToken () {
-    const { userToken: token } = await this.cache.getCached()
-    if (!token) return
-    const until = new Date(token.NotAfter)
-    const dn = Date.now()
-    const remainingMs = until - dn
-    const valid = remainingMs > 1000
-    return { valid, token: token.Token, data: token }
+  async setCachedToken (data) {
+    await this.cache.setCachedPartial(data)
   }
 
-  async getCachedXstsToken (relyingParty) {
-    const key = createHash(relyingParty)
-    const { [key]: token } = await this.cache.getCached()
-    if (!token) return
-    const until = new Date(token.expiresOn)
-    const dn = Date.now()
-    const remainingMs = until - dn
-    const valid = remainingMs > 1000
-    return { valid, token: token.XSTSToken, data: token }
-  }
+  async getCachedTokens (relyingParty) {
+    const cachedTokens = await this.cache.getCached()
 
-  async setCachedUserToken (data) {
-    await this.cache.setCachedPartial({ userToken: data })
-  }
+    const xstsHash = createHash(relyingParty)
+
+    const tokens = {
+      userToken: { valid: false },
+      titleToken: { valid: false },
+      deviceToken: { valid: false },
+      xstsToken: { valid: false }
+    }
+
+    if (cachedTokens.userToken && checkIfValid(cachedTokens.userToken.NotAfter)) tokens.userToken = { valid: true, token: cachedTokens.userToken.Token, data: cachedTokens.userToken }
+    if (cachedTokens.titleToken && checkIfValid(cachedTokens.titleToken.NotAfter)) tokens.titleToken = { valid: true, token: cachedTokens.titleToken.Token, data: cachedTokens.titleToken }
+    if (cachedTokens.deviceToken && checkIfValid(cachedTokens.deviceToken.NotAfter)) tokens.deviceToken = { valid: true, token: cachedTokens.deviceToken.Token, data: cachedTokens.deviceToken }
+    if (cachedTokens[xstsHash] && checkIfValid(cachedTokens[xstsHash].expiresOn)) tokens.xstsToken = { valid: true, data: cachedTokens[xstsHash] }
```

Comment:
the readability here can be improved with a for loop, eg
```js
  const result = {}
  for (const token of ['userToken', 'titleToken', 'deviceToken']) {
    const cached = cachedTokens[token]
    result[token] = cached && checkIfValid(cached.NotAfter)
      ? { valid: true, token: cached.Token, data: cached }
      : { valid: false }
  }
  result.xstsToken = cachedTokens[xstsHash] && checkIfValid(cachedTokens[xstsHash].expiresOn)
    ? { valid: true, data: cachedTokens[xstsHash] }
    : { valid: false }
```

Reply:
Thanks, just going to yoink that